src/components/VmStatus/VmStatus.js

```diff
@@ -78,46 +131,47 @@ const StateError = ({ children, ...props }) => (
 );
 
 export const VmStatuses = props => {
-  const { vm, launcherPod, importerPods, migration } = props;
-  const statusDetail = getVmStatusDetail(vm, launcherPod, importerPods, migration);
-  if (importerPods && importerPods.length > 1) {
-    switch (statusDetail.status) {
-      case VM_STATUS_IMPORTING:
-      case VM_STATUS_IMPORT_ERROR:
-        return (
-          <React.Fragment>
-            {importerPods.map(pod => (
-              <div key={getId(pod)}>
-                <VmStatus {...props} importerPods={[pod]} verbose />
-              </div>
-            ))}
-          </React.Fragment>
-        );
-      default:
+  const { vm, launcherPod, cdiPods, migration, dataVolumes } = props;
+
+  if (dataVolumes) {
+    const vmDataVolumes = dataVolumes.filter(
+      dv =>
+        !!getVolumes(vm)
+          .filter(volume => volume.dataVolume)
+          .find(volume => volume.dataVolume.name === getName(dv) && getNamespace(vm) === getNamespace(dv))
+    );
+
+    const statuses = vmDataVolumes.map(vmDv => getVmStatusDetail(vm, launcherPod, cdiPods, migration, [vmDv]));
+    const diskStatuses = statuses.filter(
+      status => status.status === VM_STATUS_PREPARING_DISKS || status.status === VM_STATUS_DISKS_FAILED
+    );
+
+    if (diskStatuses.length > 0) {
+      return diskStatuses.map(diskStatus => (
+        <div key={diskStatus.diskName}>{showVmStatus(vm, launcherPod, diskStatus, true)}</div>
+      ));
     }
   }
-  return (
-    <div key={importerPods && importerPods.length > 0 ? getId(importerPods[0]) : '6d0c77-has-no-importer-pods'}>
-      <VmStatus {...props} />
-    </div>
-  );
+
+  return <VmStatus {...props} />;
 };
 
 VmStatuses.defaultProps = {
   launcherPod: undefined,
-  importerPods: undefined,
+  cdiPods: undefined,
   migration: undefined,
+  dataVolumes: undefined,
 };
 
 VmStatuses.propTypes = {
   vm: PropTypes.object.isRequired,
   launcherPod: PropTypes.object,
-  importerPods: PropTypes.array,
+  cdiPods: PropTypes.array,
   migration: PropTypes.object,
+  dataVolumes: PropTypes.array,
 };
 
-export const VmStatus = ({ vm, launcherPod, importerPods, migration, verbose }) => {
-  const statusDetail = getVmStatusDetail(vm, launcherPod, importerPods, migration);
+const showVmStatus = (vm, launcherPod, statusDetail, verbose) => {
```

Comment:
this can be a component. it certainly behaves like one

Reply:
showVmStatus function is now VmStatusDetail component.